inductiva/fluids/scenarios/coastal_area/output.py

```diff
@@ -0,0 +1,242 @@
+
+
+def read_swash_output_file(file_path: str) -> Dict[str, np.ndarray]:
+    """Reads a SWASH output file.
+    
+    The output is read as a dictionary, where the keys are the names of the
+    variables and the values are the corresponding arrays.
+    """
+
+    try:
+        # Read file contents
+        data_dict = scipy.io.loadmat(file_path)
+
+    except:
+        raise ValueError(f"Error reading file '{file_path}'.") from None
+
+    return data_dict
+
+
+def read_swash_grid_positions_file(file_path: str) -> Tuple[np.ndarray]:
+    """Reads a SWASH output file containing grid positions."""
+
+    # Read grid positions
+    grid_positions_dict = read_swash_output_file(file_path)
+
+    # Get grids with x and y coordinates
+    x_grid = grid_positions_dict["Xp"].transpose()
+    y_grid = grid_positions_dict["Yp"].transpose()
+
+    # Get arrays with x and y coordinates
+    x_array = x_grid[:, 0]
+    y_array = y_grid[0, :]
+
+    return x_array, y_array
+
+
+def read_swash_quantity_file(file_path: str, quantity: str,
+                             grid_positions_file_path: str):
+    """Reads a SWASH output file containing a quantity."""
+
+    # Read grid positions
+    x_array, y_array = read_swash_grid_positions_file(grid_positions_file_path)
+
+    # Read data file
+    data_dict = read_swash_output_file(file_path)
+
+    # Convert data dictionary to lists of times and data arrays
+
+    # The list of times is built from the keys of the dictionary:
+    # the keys end with a number, in format 123456_789, identifying the time
+    # instant 12 hours, 34 minutes, 56 seconds and 789 milliseconds; we
+    # strip the times from the keys and convert them to seconds.
+
+    # The list of data arrays is build from the values of the dictionary:
+    # the values correspond to the data values at each of the time instants
+    # represented by the corresponding keys; we read each value as a 2D
+    # tensor with indices (x, y).
+
+    time_list = []
+    data_list = []
+
+    for key, data_values in data_dict.items():
+        # Check if key starts with the quantity label attributed by SWASH
+        if key.startswith(QUANTITY_SWASH_LABELS[quantity].capitalize()):
+            # Parse key string to get time in milliseconds
+            time_ms = key[len(QUANTITY_SWASH_LABELS[quantity]):].replace(
+                "_", "")
+
+            # Convert time to seconds
+            time_s = float(time_ms[:2]) * 3600. + \
+                    float(time_ms[2:4]) * 60. + \
+                    float(time_ms[4:6]) + \
+                    float(time_ms[6:]) / 1000.
+
+            # The values are transposed in the file, so we transpose them
+            # again for consistency
+            transposed_data_values = data_values.transpose()
+
+            time_list.append(time_s)
+            data_list.append(transposed_data_values)
+
+    return (x_array, y_array), time_list, data_list
+
+
+def _render_quantity_data(x_array: np.ndarray,
```

Comment:
why not include these functions as methods of the class?

Reply:
As @IvanPombo said, part of these functions can be used to plot results of other SWASH simulations, so I didn't want to tie them to the class introduced here.

Reply:
Ah makes sense, then shouldn't it be elsewhere?

Reply:
I think so, but I don't think this is the moment to make a decision about that. See [my comment](https://github.com/inductiva/inductiva/pull/396#issuecomment-1651421002) below.